docs/branching.md

```diff
@@ -0,0 +1,35 @@
+# Git Branching
+
+The following is a modified version of the gitflow branching strategy described in <https://nvie.com/posts/a-successful-git-branching-model/>
+
+## Main Branches
+
+- **master** - `origin/master` is the main branch where the source code of HEAD always reflects a production-ready state.
+- **develop** - `origin/develop` is the branch where the source code of HEAD always reflects a state with the latest delivered development changes for the next release.
+- **next** - `origin/next` may contain consensus-breaking changes.
+- **release/X.Y.Z.A.n** is the release branch.
+
+When the source code in the develop branch reaches a stable point and is ready to be released, a release branch is created as `release/X.Y.Z.A.n` (see [release-process.md](./release-process.md)).
+After release, the following will happen:
+
+- `release/X.Y.Z.A.n` branch is merged back to `origin/master`.
```

Comment:
I have also seen tagged [releases](https://github.com/stacks-network/stacks-core/releases/) and wondered: do they `tag` the commit the merges to `master` or the tip of the `release/X.Y.Z...` branch?

Reply:
when the workflow run, the tag hash points to the release branch. in the past, when we released off of master the tag would point to a commit on the master branch.